gammapy/stats/feldman_cousins.py

```diff
@@ -0,0 +1,442 @@
+
+    Parameters
+    ----------
+    mu : double
+        Mean of the Gaussian
+    sigma : double
+        Width of the Gaussian
+    x_bins : array-like
+        Bins in x
+    alpha : double
+        Desired confidence level
+
+    Returns
+    -------
+    (x_min, x_max) : tuple of floats
+        Acceptance interval
+    """
+
+    from scipy import stats
+
+    dist = stats.norm(loc=mu, scale=sigma)
+
+    x_bin_width = x_bins[1] - x_bins[0]
+
+    p = []
+    r = []
+
+    for x in x_bins:
+        p.append(dist.pdf(x)*x_bin_width)
+        # This is the formula from the FC paper
+        if mu == 0 and sigma == 1:
+            if x < 0:
+                r.append(np.exp(mu*(x-mu*0.5)))
+            else:
+                r.append(np.exp(-0.5*np.power((x-mu),2)))
+        # This is the more general formula
+        else:
+            # Implementing the boundary condition at zero
+            muBest     = max(0, x)
+            probMuBest = stats.norm.pdf(x, loc=muBest, scale=sigma)
+            if probMuBest == 0.0:
+                r.append(0.0);
+            else:
+                r.append(p[-1]/probMuBest)
+
+    p = np.asarray(p)
+    r = np.asarray(r)
+
+    if sum(p) < alpha:
+        log.info("Bad choice of x-range for this mu!")
+        log.info("Not enough probability in x bins to reach confidence level!")
+
+    rank = stats.rankdata(-r, method='dense')
+
+    index_array = np.arange(x_bins.size)
+
+    rank_sorted, index_array_sorted = zip(*sorted(zip(rank, index_array)))
+
+    index_min = index_array_sorted[0]
+    index_max = index_array_sorted[0]
+
+    p_sum = 0
+
+    for i in range(len(rank_sorted)):
+        if index_array_sorted[i] < index_min:
+            index_min = index_array_sorted[i]
+        if index_array_sorted[i] > index_max:
+            index_max = index_array_sorted[i]
+        p_sum += p[index_array_sorted[i]]
+        if p_sum >= alpha:
+            break
+
+    return x_bins[index_min], x_bins[index_max] + x_bin_width
+
+
+def fc_find_acceptance_region_poisson(mu, background, x_bins, alpha):
+    """Analytically find the acceptance region (x_min, x_max) for Poisson
+       process with background, such that int_x_min^x_max P(x|mu)dx = alpha
+
+    Parameters
+    ----------
+    mu : double
+        Mean of the signal
+    background : double
+        Mean of the background
+    x_bins : array-like
+        Bins in x
+    alpha : double
+        Desired confidence level
+
+    Returns
+    -------
+    (x_min, x_max) : tuple of floats
+        Acceptance interval
+    """
+
+    from scipy import stats
+
+    dist = stats.poisson(mu=mu+background)
+
+    x_bin_width = x_bins[1] - x_bins[0]
+
+    p = []
+    r = []
+
+    for x in x_bins:
+        p.append(dist.pmf(x))
+        # Implementing the boundary condition at zero
+        muBest = max(0, x - background)
+        probMuBest = stats.poisson.pmf(x, mu=muBest+background)
+        if probMuBest == 0.0:
+            r.append(0.0);
```

Comment:
I think this line is actually not needed because P(n|mu_best) should never be zero. But it doesn't hurt to check for zero before division (the ROOT code does the same test).

(Semicolon removed)